src/host/topology.c

```diff
@@ -836,11 +836,13 @@ int parse_topology(struct sof *sof, struct shared_lib_table *library_table,
 			sprintf(message, "number of DAPM widgets %d\n",
 				hdr->count);
 			debug_print(message);
-			size = sizeof(struct comp_info) * hdr->count;
-			temp_comp_list = (struct comp_info *)malloc(size);
-			num_comps = hdr->count;
 
-			for (i = 0; i < hdr->count; i++)
+			num_comps += hdr->count;
```

Comment:
@jajanusz sorry for the delayed response. although harmless, this step is unnecessary because there will only ever be one DAPM_WIDGET header in topology containing all the widgets, if I am not mistaken. So the num_comps should just be equal to the hdr->count.